tools/ts_library_builder/ast_util.ts

```diff
@@ -0,0 +1,331 @@
+import { relative } from "path";
+import { readFileSync } from "fs";
+import { EOL } from "os";
+import {
+  ExportDeclaration,
+  ImportDeclaration,
+  InterfaceDeclaration,
+  JSDoc,
+  Project,
+  PropertySignature,
+  SourceFile,
+  StatementedNode,
+  ts,
+  TypeGuards,
+  VariableStatement,
+  VariableDeclarationKind
+} from "ts-simple-ast";
+
+/** Add a property to an interface */
+export function addInterfaceProperty(
+  interfaceDeclaration: InterfaceDeclaration,
+  name: string,
+  type: string,
+  jsdocs?: JSDoc[]
+): PropertySignature {
+  return interfaceDeclaration.addProperty({
+    name,
+    type,
+    docs: jsdocs && jsdocs.map(jsdoc => jsdoc.getText())
+  });
+}
+
+/** Add `@source` comment to node. */
```

Comment:
Does a URL make more sense?
I think we're mostly doing this for our imaginative users who don't have the source code checked out.

Reply:
Well, I was doing it to be able to see where it came from when chasing down issues... I think also in the future, if an interface or API pops into the generated library and we want to not "leak" it and want to mark it with `@internal`, being able to determine that in debug mode would make it easier. `@url` can make more sense/is a more standard JSDoc pragma.